### PR TITLE
Charlesmchen/oversize text message view

### PR DESCRIPTION
**Pull request by charlesmchen**
This work builds on https://github.com/WhisperSystems/Signal-iOS/pull/1904 which is not yet merged.

PTAL @michaelkirk 

screen shot 2017-03-29 at 12 19 48 pm
screen shot 2017-03-29 at 12 19 42 pm

### CODE REVIEW

File: Signal/src/view controllers/OversizeTextMessageViewController.swift [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    // MARK: Attachment
+
+    private func attachmentStream() -> TSAttachmentStream? {
+        guard message.hasAttachments() else {
+            Logger.error("\(TAG) message has no attachments.")
+            assert(false)
+            return nil
+        }
+        let attachmentIDRaw = message.attachmentIds[0]
+        guard attachmentIDRaw is String else {
```
Comment (michaelkirk) [inline Signal/src/view controllers/OversizeTextMessageViewController.swift]: nit:

or:
```
guard let attachmentID = message.attachmentIds[0] as? String else {
    Logger.error("\(TAG) message attachment id is not a string.")
    assertionFailure()
    return nil
}
```

File: Signal/src/view controllers/OversizeTextMessageViewController.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            Logger.error("\(TAG) could not load attachment.")
+            assert(false)
+            return nil
+        }
+        guard attachment.contentType == OWSMimeTypeOversizeTextMessage else {
+            Logger.error("\(TAG) attachment has unexpected content type.")
+            assert(false)
+            return nil
+        }
+        guard attachment is TSAttachmentStream else {
```
Comment (michaelkirk) [inline Signal/src/view controllers/OversizeTextMessageViewController.swift]: nit: could combine a couple of these guards to be more concise and avoid the `!`
```
guard let stream = TSAttachment.fetch(withUniqueID:attachmentID) as? TSAttachmentStream else {
...
}
```

File: Signal/src/view controllers/OversizeTextMessageViewController.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+        self.navigationItem.title = NSLocalizedString("OVERSIZE_TEXT_MESSAGE_VIEW_TITLE",
+                                                      comment: "The title of the 'oversize text message' view.")
+
+        self.view.backgroundColor = UIColor.white
+
+        let textView = UITextView()
+        textView.textColor = UIColor.black
+        textView.text = displayText()
+        textView.font = UIFont.ows_regularFont(withSize:14)
```
Comment (michaelkirk) [inline Signal/src/view controllers/OversizeTextMessageViewController.swift]: can we use dynamic type font here?

**Comment by charlesmchen** [comment]
@michaelkirk Sorry, I just saw your comments here.  I'll address them post merge.

**Comment by charlesmchen** [comment]
> I feel like we need to do more to show that [Tap for More] is interface, not content. i.e. it's actionable text, rather than something the sender sent.

Agreed.  I just haven't figured out what to do yet.  I'm open to ideas.  One thing I considered is making an oversize text adapter like the generic attachment adapter.  I actually did that already and threw it away once.  But I'm starting to think it was the right approach.

**Comment by charlesmchen** [comment]
> I feel like we need to do more to show that [Tap for More] is interface, not content. i.e. it's actionable text, rather than something the sender sent.

I'm doing this work under another trello card.

File: Signal/src/view controllers/OversizeTextMessageViewController.swift (hunk 1 above)
Comment (charlesmchen) [inline Signal/src/view controllers/OversizeTextMessageViewController.swift]: Done, nice, thank you.

File: Signal/src/view controllers/OversizeTextMessageViewController.swift (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/view controllers/OversizeTextMessageViewController.swift]: Done.

**Comment by charlesmchen** [comment]
https://github.com/WhisperSystems/Signal-iOS/commit/3765d28da9144b28cb8777f4194ab88f169d08b8

File: Signal/src/view controllers/OversizeTextMessageViewController.swift (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/view controllers/OversizeTextMessageViewController.swift]: Done.

**Pull request merged**